### PR TITLE
proxy: server: support static vc proxy

### PR DESCRIPTION
**Pull request by kubistika**
This PR adds couple of features to the proxy:
* Proxy static virtual channel without the need to implement them in freerdp/proxy
* New API for proxy plugins that a plugin to filter/log the passthrough vc data:
  * ClientChannelData - client received data from server
  * ServerChannelData - server received data from peer

Also moved up PluginUnload callback so it won't break the ABI next time when the proxy plugins API will be extended.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by kubistika** [comment]
@akallabeth sorry, was on vacation. nope, it can be merged.

File: server/proxy/modules/modules_api.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ typedef struct proxy_plugin
 	const char* name;        /* unique module name */
 	const char* description; /* module description */
 
+	BOOL (*PluginUnload)();
```
Comment (akallabeth) [inline server/proxy/modules/modules_api.h]: should this really be a function pointer with variable arguments?

**Pull request merged**